karyon3-eureka/src/main/java/com/netflix/karyon/eureka/KaryonHealthCheckHandler.java

```diff
@@ -0,0 +1,81 @@
+package com.netflix.karyon.eureka;
+
+import javax.inject.Singleton;
+
+import com.google.inject.Inject;
+import com.netflix.appinfo.HealthCheckHandler;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+import com.netflix.governator.LifecycleListener;
+import com.netflix.karyon.healthcheck.HealthCheck;
+import com.netflix.karyon.healthcheck.HealthChecks;
+
+/**
+ * Eureka HealthCheckHandler implementation that combines Karyon's HealthCheck with
+ * the injector lifecycle to auto-register with Eureka once the injector has been created
+ * and the HealthCheck is healthy.
+ * 
+ * @author elandau
+ */
+@Singleton
+public class KaryonHealthCheckHandler implements HealthCheckHandler, LifecycleListener {
```

Comment:
This is a bit more generic than just for eureka integration. It might make sense for users to be able to decouple Karyon's health state from the injector state, for cases when applications need to asynchronously perform warm up after the injector has returned, before it is ready to be Started.